### PR TITLE
readme: add installation instructions

### LINKED ISSUE #1641: go get <package> not working
> go get github.com/google/go-github/tree/v32.1.0/github 

or

> go get github.com/google/go-github/v32/github

package github.com/google/go-github/tree/v32.1.0/github: cannot find package "github.com/google/go-github/tree/v32.1.0/github" in any of:
        /usr/local/go/src/github.com/google/go-github/tree/v32.1.0/github (from $GOROOT)
        /Users/<username>/go/src/github.com/google/go-github/tree/v32.1.0/github (from $GOPATH)

### PR DESCRIPTION
**Pull request by DarthBenro008**
- Fixes #1641 
- Added Installation Instructions
- Added version badge sorted by latest semver

### CODE REVIEW

File: README.md
Comment (gmlewis) [inline README.md:3]: What if we make this a link like the other banners below?  We could link it to the releases page, like this:

```
[![GitHub release (latest SemVer)](https://img.shields.io/github/v/release/google/go-github?sort=semver)]
(https://github.com/google/go-github/releases)
```

File: README.md
Comment (gmlewis) [inline README.md:39]: How about adding this to the end of this section so that the question never has to come up again?

```
Finally, to use the top-of-trunk version of this repo, use the following command:

` ` `bash
go get -u github.com/google/go-github/v33@master
` ` `
```

(Obviously, please fix the backticks.)

File: README.md
Comment (gmlewis) [inline README.md:24]:
```suggestion
go-github is compatible with modern Go releases in module mode, with Go installed:
```

(Please remove the extra space before the colon.)

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @jnewland !
Just a few minor tweaks, please, and then I think we will be ready to merge after we get a second LGTM.

**Comment by DarthBenro008** [comment]
@gmlewis Thank you for you feedback! I have tweaked accordingly!

**Review by gmlewis — APPROVED** [review]
Looks great, @DarthBenro008 ... thank you!
LGTM.
Merging.

File: README.md
Comment (dmitshur) [inline README.md:27]: We should not recommend the `-u` flag in module mode. It was needed in GOPATH mode because there was no other mechanism to ensure compatible versions of dependencies would be used, but in module mode the go.mod file provides well-tested minimum dependency version requirements.

```suggestion
go get github.com/google/go-github/v33
```

File: README.md
Comment (dmitshur) [inline README.md:43]: Similarly here:

```suggestion
go get github.com/google/go-github/v33@master
```

**Comment by gmlewis** [comment]
Thank you for the feedback, @dmitshur!
I'll put together a quick PR to remove the `-u` flags.

**Pull request merged**
